vault/resource_pki_backend_role.go

```diff
@@ -0,0 +1,380 @@
+package vault
+
+import (
+	"fmt"
+	"log"
+	"regexp"
+	"strings"
+
+	"github.com/hashicorp/terraform/helper/schema"
+	"github.com/hashicorp/vault/api"
+)
+
+var (
+	pkiBackendRoleBackendFromPathRegex = regexp.MustCompile("^(.+)/roles/.+$")
+	pkiBackendRoleNameFromPathRegex    = regexp.MustCompile("^.+/roles/(.+)$")
+)
+
+func pkiBackendRoleResource() *schema.Resource {
+	return &schema.Resource{
+		Create: pkiBackendRoleWrite,
+		Read:   pkiBackendRoleRead,
+		Update: pkiBackendRoleWrite,
+		Delete: pkiBackendRoleDelete,
+		Exists: pkiBackendRoleExists,
+		Importer: &schema.ResourceImporter{
+			State: schema.ImportStatePassthrough,
+		},
+
+		Schema: map[string]*schema.Schema{
+			"role": {
+				Type:        schema.TypeString,
+				Required:    true,
+				ForceNew:    true,
+				Description: "Name of the role.",
+			},
+			"ttl": {
+				Type:        schema.TypeString,
+				Optional:    true,
+				Computed:    true,
+				Description: "The TTL period of certificates issued using this role, provided as a duration.",
+			},
+			"max_ttl": {
+				Type:        schema.TypeString,
+				Optional:    true,
+				Computed:    true,
+				Description: "The maximum TTL of certificates issued using this role, provided as a duration.",
+			},
+			"allow_localhost": {
+				Type:        schema.TypeBool,
+				Optional:    true,
+				Default:     true,
+				Description: "Specifies if clients can request certificates for localhost as one of the requested common names.",
+			},
+			"allowed_domains": {
+				Type:     schema.TypeList,
```

Comment:
If the order doesn't matter here, I might suggest a `schema.TypeSet` instead.

Reply:
Sure, order does not matter, changed.